algorithmic_efficiency/checkpoint_utils.py

```diff
@@ -136,7 +136,8 @@ def save_checkpoint(framework: str,
     model_params = model_params.state_dict()
     optimizer_state_dict = {}
     for key in optimizer_state.keys():
-      optimizer_state_dict[key] = optimizer_state[key].state_dict()
+      if hasattr(optimizer_state[key], 'state_dict'):
+        optimizer_state_dict[key] = optimizer_state[key].state_dict()
```

Comment:
I think we should do a logging.warning here (on the first call?) that we are ignoring `key`, so the user has a chance to know.

Reply:
Good idea, I have added a warning.